internal/server/data/data.go

```diff
@@ -78,44 +77,69 @@ func NewDB(dbOpts NewDBOptions) (*DB, error) {
 // DB wraps the underlying database and provides access to the default org,
 // and settings.
 type DB struct {
-	*gorm.DB // embedded for now to minimize the diff
+	DB *sql.DB
 
 	DefaultOrg *models.Organization
 	// DefaultOrgSettings are the settings for DefaultOrg
 	DefaultOrgSettings *models.Settings
 }
 
 func (d *DB) Close() error {
-	sqlDB, err := d.DB.DB()
-	if err != nil {
-		return fmt.Errorf("failed to get database conn to close: %w", err)
-	}
-	return sqlDB.Close()
+	return d.DB.Close()
 }
 
 func (d *DB) SQLdb() *sql.DB {
-	sqlDB, err := d.DB.DB()
-	if err != nil {
-		panic("DB must have an sql.DB ConnPool")
-	}
-	return sqlDB
-}
-
-func (d *DB) DriverName() string {
-	return d.Dialector.Name()
+	return d.DB
 }
 
 func (d *DB) Exec(query string, args ...any) (sql.Result, error) {
-	db := d.DB.Exec(query, args...)
-	return driver.RowsAffected(db.RowsAffected), db.Error
+	start := time.Now()
+	query = rewriteQueryPlaceholders(query, len(args))
+	result, err := d.DB.Exec(query, args...)
+
+	// TODO: result.RowsAffected()
+	logQuery(query, err, start, -1)
+	return result, err
 }
 
 func (d *DB) Query(query string, args ...any) (*sql.Rows, error) {
-	return d.DB.Raw(query, args...).Rows()
+	start := time.Now()
+	query = rewriteQueryPlaceholders(query, len(args))
+	rows, err := d.DB.Query(query, args...)
+	logQuery(query, err, start, -1)
+	return rows, err
+
 }
 
 func (d *DB) QueryRow(query string, args ...any) *sql.Row {
-	return d.DB.Raw(query, args...).Row()
+	start := time.Now()
+	query = rewriteQueryPlaceholders(query, len(args))
+	row := d.DB.QueryRow(query, args...)
+	logQuery(query, row.Err(), start, -1)
+	return row
+}
+
+func rewriteQueryPlaceholders(query string, num int) string {
+	var counter int
+	var buf strings.Builder
+	buf.Grow(len(query))
+
+	for _, r := range query {
+		if r != '?' {
```

Comment:
not all ? are necessarily safe to assume as placeholders unless you're parsing the sql.

Reply:
Which cases are not safe to replace?  If someone is putting user input in this part of the query then it's a potential SQL injection, so that should never be the case. When would we write a `?` into a query that was not a placeholder?

Reply:
I had a read over the gorm source code, and from what I can tell this is all it was doing (albeit with a ton more abstraction and complexity that makes it hard to see). It's possible I read it wrong.

Reply:
My understanding is that this is safe because:
1. Any user input values can not be in that part of the query
2. Postgres identifiers can not contain a ? (https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS)
3. Even if we did run into a problem that we haven't accounted for, it would be from a query we wrote in the source code. So we'd notice it in CI when someone wrote a test for it. So worst case we could address the problem then. And maybe at that point we'll already have replaced all of our queries with `$n` literals, which will allow us to delete this code, and it won't be a problem.

Reply:
We chatted about this a bit offline this week. It is possible that we write a query that uses something with `LIKE = '?..'`, however we can always rewrite that query to `LIKE = ?` with a value of `"?..."`, and since that query needs to be part of the source (not user input), we should catch that problem in tests pretty easily.